src/natalie_parser/lexer.cpp

```diff
@@ -103,4 +1406,16 @@ void InterpolatedStringLexer::tokenize_interpolation(SharedPtr<Vector<Token>> to
     tokens->push(Token { Token::Type::EvaluateToStringEnd, m_file, m_line, m_column });
 }
 
+char InterpolatedStringLexer::current_char() {
+    if (m_index >= m_size)
+        return 0;
+    return m_input->at(m_index);
+}
+
+char InterpolatedStringLexer::peek() {
+    if (m_index + 1 >= m_size)
+        return 0;
+    return m_input->at(m_index + 1);
+}
```

Comment:
Let's move current_char() and peek() back to the header just to ensure they get inlined.